### DISCUSSION TITLE
How to update fork's pull request HEAD from the Github API?

### DISCUSSION THREAD

**Discussion by joao-paulo-parity** [community/community#24767] [Apps, API and Webhooks]
<p><strong>Note:</strong> The PR has “allow edits from maintainers” enabled.</p>
<p>I am trying to use the <code>PATCH /repos/{owner}/{repo}/git/refs/{ref}</code> API to update a fork’s pull request HEAD through the API.</p>
<p>If it were not a fork, I’d be able to use the <code>heads/{branch}</code> ref.</p>
<p>I have tried to use the <code>pull/{number}/head</code> ref but I can’t because it’s read-only.</p>
<p>I have tried to target the fork’s ref (i.e. <code>/repos/{fork_owner}/{fork_repo}/git/refs/heads/{fork_branch}</code>) but then I get access denied.</p>
<p>Not sure what to try next.</p>

**Comment by nethgato** [comment]
<p>Hey there <a class="mention" href="/joao-paulo-parity">@joao-paulo-parity</a> :wave:</p>
<p>Just a couple questions here to help move things forward…</p>
<p>Is this fork from a private repository, and does it happen to be from within an Organization, or from a User account?</p>
<p>The relationship between the fork and it’s “parent,” will not be able to be released by API action alone and would likely require a manual action from GitHub staff.</p>
<p>If you could let us know the particulars of the repo and the fork you’re trying to work from, that would be excellent!</p>

**Comment by joao-paulo-parity** [comment, marked as answer]
<p>Hello. I’ve asked this almost a week ago so I have a workaround for the the problem now: I’m creating a temporary ref, pulling it, then deleting it after.</p>
<p><a href="https://github.com/paritytech/parity-processbot/pull/287/files#diff-183303ed774acdd18947d78921d79ea350c6d731268d84b6572d785e44233661R297" rel="noopener nofollow ugc">Here’s the code for the workaround</a>.</p>
<p>I’m going through all of this trouble because I want the bot’s commits to be signed and verified; and the only way I found of doing that is by creating commits through the API.</p>
